chore: clean up interfaces (#1562)
Layr-Labs/eigenlayer-contracts: docs/multichain/destination/CertificateVerifier.md

/**
 * @notice A struct that contains information about a single operator for an ECDSA signing key
 * @param pubkey The address of the signing ECDSA key of the operator and not the operator address itself.
 * @param weights The weights of the operator for a single operatorSet
 *
 * @dev The `weights` array can be defined as a list of arbitrary stake types. For example,

>  *      it can be [slashable_stake, delegated_stake, strategy_i_stake, ...]. Each stake type is an index in the array

Reviewer: In addition to this, would be nice to also specify why an AVS would choose to define their stake types a certain way. E.g. why would an AVS want to specify `strategy_i_stake` specifically separate from the `slashable_stake`? Worth putting down / pointing to a few examples for AVSs to use as reference

}

/**
 * @notice updates the operatorSet with the operator table (ie. stake weights) and its configuration
 * @param operatorSet the operatorSet to update the operator table for
 * @param referenceTimestamp the timestamp at which the operatorInfos (ie. operator table) was sourced

>  * @param operatorInfos the operatorInfos to update the operator table with

Author: it's above?

 * @param operatorSetConfig the configuration of the operatorSet, which includes the owner and max staleness period

>  * @dev Only callable by the `OperatorTableUpdater`, the function on which is permissionless

Reviewer 2: ```suggestion
 * @dev Only callable by the `OperatorTableUpdater`, the function on which is permissionless
```
@dev This function can only be called by the OperatorTableUpdater contract, which is itself permissionless to call.”

>  * @dev The `referenceTimestamp` must correspond to a reference timestamp for a globalTableRoot stored in the `OperatorTableUpdater`

Author: https://github.com/Layr-Labs/eigenlayer-contracts/pull/1562/commits/9d0dd05446a914c9a59b6ebc6bf74162df489595

}

/**
 * @notice verifies a certificate against the operator table for a given reference timestamp
 * @param operatorSet the operatorSet that the certificate is for
 * @param cert a certificate
 * @return totalSignedStakeWeights total stake weight that signed the certificate for each stake type. Each
 * index corresponds to a stake type in the `weights` array in the `ECDSAOperatorInfo` struct
 * @return signers array of addresses that signed the certificate
 * @dev This function DOES NOT support smart contact signatures
 * @dev The `referenceTimestamp` in the `ECDSACertificate` is used to determine the operator table to use for the verification

>  * @dev It is up to the AVS to handle race conditions for certificates against stale or new operator tables

Author: https://github.com/Layr-Labs/eigenlayer-contracts/pull/1562/commits/53748b8c014f341f79bb9931128ad96c07262a4b

 * @notice A struct that contains information about a single operator for a given BN254 operatorSet
 * @param pubkey The G1 public key of the operator
 * @param weights The weights of the operator for a single operatorSet
 *
 * @dev The `weights` array is as a list of arbitrary stake types. For example,
 *      it can be [slashable_stake, delegated_stake, strategy_i_stake, ...]. Each stake type is an index in the array
 *
 * @dev It is up to the AVS to define the `weights` array, which is used by the `IBN254CertificateVerifier` to verify Certificates
 */
struct BN254OperatorInfo {
    BN254.G1Point pubkey;
    uint256[] weights;
}

/**

>  * @notice A struct that contains information about all operators for a given BN254operatorSet

Reviewer: ```suggestion
 * @notice A struct that contains information about all operators for a given BN254OperatorSet
```

}

/**
 * @notice verifies a certificate against the operator table for a given reference timestamp
 * @param operatorSet the operatorSet that the certificate is for
 * @param cert a certificate
 * @return totalSignedStakeWeights total stake weight that signed the certificate for each stake type. Each
 *         index corresponds to a stake type in the `weights` array in the `BN254OperatorSetInfo` struct
 * @dev The `referenceTimestamp` in the `BN254Certificate` is used to determine the operator table to use for the verification
 * @dev It is up to the AVS to handle race conditions for certificates against stale or new operator tables
 * @dev Reverts if the certificate's `referenceTimestamp` is too stale with respect to the `maxStalenessPeriod` of the operatorSet
 * @dev This function is *non-view* because it caches non-signing operator info upon a successful certificate verification. See `getNonsignerOperatorInfo` for more details
 */
function verifyCertificate(
    OperatorSet memory operatorSet,
    BN254Certificate memory cert
) external returns (uint256[] memory totalSignedStakeWeights);
```

> Verifies a BN254 certificate by checking the aggregated signature against the operator set's aggregate public key. *Note: This function is non-view because the non-signers are cached in storage.*. See [cachingMechanism](#caching-mechanism) for more information. **The `referenceTimestamp` of the certificate is used to get the operators and their stake weights for certificate verification. It is not the timestamp at which the certificate was created.**

Reviewer: ```suggestion
Verifies a BN254 certificate by checking the aggregated signature against the operator set's aggregate public key. *Note: This function is non-view because the non-signers are cached in storage.*. See [cachingMechanism](#caching-mechanism) for more information. **The `referenceTimestamp` of the certificate is the time at which the operator table for a given certificate was sourced. It is not the timestamp at which the certificate was created.**
```

By using parallel language, we make it much more clear what the `referenceTimestamp` is and isn't. Feel free to reword as well, but I suggest we say it **is** this, and **is not** that, rather than use different verbs

Author: https://github.com/Layr-Labs/eigenlayer-contracts/pull/1562/commits/9986609cd63e31f776709ff1d18a7c497c6cddcd

/**
 * @notice A witness for an operator, used to identify the non-signers for a given certificate
 * @param operatorIndex the index of the nonsigner in the `BN254OperatorInfo` tree

>  * @param operatorInfoProofs merkle proof of the nonsigner at the index. Empty if the non-signing operator is   already stored from a previous verification

Reviewer 2: for the Empty, might be more clear to say Leave empty if.....

>  * @param operatorInfoProofs merkle proof of the nonsigner at the index. Empty if the non-signing operator is   already stored from a previous verification

Reviewer: ```suggestion
 * @param operatorInfoProofs merkle proof of the nonsigner at the index. Empty if the non-signing operator is already stored from a previous verification
```

 * @param operatorInfo the `BN254OperatorInfo` for the operator. Empty if the non-signing operator is already stored from a previous verification
 * @dev Non-signing operators are stored in the `BN254CertificateVerifier` upon the first successful certificate verification that includes a merkle proof for the non-signing operator. 

>  *  This is done to avoid the need for resupplying proofs of non-signing operators for each certificate verification at a given reference timestamp

Author: https://github.com/Layr-Labs/eigenlayer-contracts/pull/1562/commits/55806cd48f3d29166d286956c35f16f0959c9a60

>  *  This is done to avoid the need for resupplying proofs of non-signing operators for each certificate verification at a given reference timestamp

Author: https://github.com/Layr-Labs/eigenlayer-contracts/pull/1562/commits/55806cd48f3d29166d286956c35f16f0959c9a60
